lib/ui/window/platform_configuration.cc

```diff
@@ -284,15 +285,33 @@ void PlatformConfigurationNativeApi::SetNeedsReportTimings(bool value) {
       ->SetNeedsReportTimings(value);
 }
 
+namespace {
+Dart_Handle HandlePlatformMessage(
+    UIDartState* dart_state,
+    const std::string& name,
+    Dart_Handle data_handle,
+    const fml::RefPtr<PlatformMessageResponse>& response) {
+  if (Dart_IsNull(data_handle)) {
+    return dart_state->HandlePlatformMessage(
+        std::make_unique<PlatformMessage>(name, response));
+  } else {
+    tonic::DartByteData data(data_handle);
+    const uint8_t* buffer = static_cast<const uint8_t*>(data.data());
+    return dart_state->HandlePlatformMessage(std::make_unique<PlatformMessage>(
+        name, fml::MallocMapping::Copy(buffer, data.length_in_bytes()),
+        response));
+  }
+}
+}  // namespace
+
 Dart_Handle PlatformConfigurationNativeApi::SendPlatformMessage(
     const std::string& name,
     Dart_Handle callback,
     Dart_Handle data_handle) {
   UIDartState* dart_state = UIDartState::Current();
 
   if (!dart_state->platform_configuration()) {
-    return tonic::ToDart(
-        "Platform messages can only be sent from the main isolate");
+    return tonic::ToDart("SendPlatformMessage only works on the root isolate.");
```

Comment:
This should probably mention the correct API to use instead.

It seems like it's a shame that we have two separate API paths for this - I can imagine a developer being confused/surprised to find that they're on a background isolate (because the code they're hacking on is getting called from `compute` by some other far off code, or because it's being invoked from a plugin that used a background isolate and they didn't realize it).

I think if we were writing this all from scratch we'd probably try to create a unified API. I'm not really sure what that would look like. I haven't looked at the framework PR yet and maybe that abstracts away this difficulty which is fine.

Reply:
Done.  Yea I agree.  I considered migrating everyone to the port implementation but I had concerns about potentially changing performance and it's a higher hurdle for reviewing.  It's better to investigate it separately.